src/pages/index.vue

```diff
@@ -1,81 +1,114 @@
-<template>
-  <HelloWorld msg="Vite Vue TS" />
+<script setup lang="ts">
+import Button from 'primevue/button'
+import Dropdown from 'primevue/dropdown'
+import InputText from 'primevue/inputtext'
+import { useRouter } from 'vue-router'
+import 'primeicons/primeicons.css'
+import {
+  searchParam,
+  pageSize,
+  pageSizes,
+  displayData,
+  displayedPokemon,
+  currentPage,
+  pokeFavorites,
+  favorites
+} from './variables'
+import { searchButtonHandler, getData } from '@/helpers/homepageHelpers'
+import { saveFavorites } from '@/helpers/storeFavorite'
 
-  <WelcomeItem>
-    <template #icon>
-      <DocumentationIcon />
-    </template>
-    <template #heading>Documentation</template>
+const router = useRouter()
 
-    Vue's
-    <a href="https://vuejs.org/" target="_blank" rel="noopener">official documentation</a>
-    provides you with all information you need to get started.
-  </WelcomeItem>
+const updateDisplayed = async () => {
+  displayData.value = await getData(currentPageSize.value, currentPage.value)
+  displayedPokemon.value = displayData.value.results.map((pokemon) => {
+    return {
+      name: pokemon.name,
+      id: +pokemon.url.split('pokemon/')[1].slice(0, -1),
+      url: `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${pokemon.url.split('pokemon/')[1].slice(0, -1)}.png`
+    }
+  })
+}
 
-  <WelcomeItem>
-    <template #icon>
-      <ToolingIcon />
-    </template>
-    <template #heading>Tooling</template>
+const currentPageSize = computed(() => {
+  const num: number = +pageSize.value
+  // console.log(num)
+  if (isNaN(num)) {
```

Comment:
There's actually two different `isNaN` functions available to use, [read here for more info](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN), but the tldr is you might want to use `Number.isNaN` instead